packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -456,6 +464,13 @@ export class PipelineEditor extends React.Component<
     }
     const app_data = node.app_data;
 
+    if (app_data.filename !== propertySet.filename) {
+      app_data.filename = propertySet.filename;
+      node.label = propertySet.filename
+        .replace(/^.*[\\/]/, '')
+        .replace(/\.[^/.]+$/, '');
```

Comment:
This is stripping off the directory name prefix and any suffix, leaving just the base filename.  Two questions...
1. Are there library methods that can do that is typescript (similar to Python `os.path`)?
2. If not, would you mind adding a simple comment to this effect so folks new to this can more easily maintain it?

I know folks will get this after a bit, but this is one of those places where a few words could save someone getting unnecessarily lost in the regex weeds.

Reply:
if this is just getting the filename then you should use `path.basename` we already use that on line 802 in this file (in the export function)

Reply:
also on a bit of looking we should probably make this change to use the `path` lib in a few other places in this file (it seems you took this regex from the add function).

Reply:
@kevin-bates there are libraries that do that but currently none are included as a dependency in the pipeline-editor package. since it is a short method similar code and it is being called in multiple places i have moved it into the `Utils` class (and updated and documented accordingly).

Reply:
i stand corrected. the library is already imported and being. will update accordingtl

Reply:
done https://github.com/elyra-ai/elyra/pull/867/commits/4dfe3be0a09a5964f7d5e9c6b55e888571b72314